### DISCUSSION TITLE
Feature Request: variables for panel options

### DISCUSSION THREAD

**Discussion by jufy111** [grafana/grafana#55008] [Dashboards, Panels & Visualizations]
It would be great to be able to use variables for the panel options.

For example, in a dashboard full of timeseries graphs, being able to assign a variable as the "axis width" or "legend width" would make it easy to adjust the width of the legend or axis for all panels in the dashboard by simply changing the variable. 
This would also add the functionality of being able to hide the code for all meetings by simply setting the width to zero.

**Comment by jaxels20** [comment]
+1

**Comment by eeik** [comment]
I'd like to +1 this feature request and extend it to also ensure the following use case is considered. I discovered this configuration gap and open feature when trying to adapt a panel's vertical axis to accommodate a data driven range where the "soft min" / "soft max" concepts don't work as fixed values.

Not only would it be important to be able to assign these min/max values to something variable, they should also be able to be derived from a formula given the actual data within the viewport. For instance, one panel averages around 150,000 with variation roughly +/- 2% (147,000 - 153,000) so I'd like to set my soft min to be 2% below the expected range.

This same panel when toggling the input filter for the dashboard, may then shift to 380,000 w/ the same +/- 2% range (372,400 - 387,600).

This panel configuration might look like ... `soft min = 0.98 * min(data_series)`

If this were configurable as variables for the dashboard, that might work as well, but it creates a need to manage a set of variables vs. allowing the calculations to be truly dependent upon the data actively shown within the panel view which seems more elegant than managing a range of external variables.

**Comment by Hivanoe** [comment]
For me it also would be fine to use a variable for "soft min" and "soft max" axis scale

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). We are migrating this discussion to an issue and closing the discussion. The issue is #82938. Feel free to continue the discussion around this there. Thank you!
